Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -455,6 +464,43 @@ public final class OAuth2Client {
         }
     }
     
+    private func revokeAll(_ token: Token, completion: @escaping (Result<Void, OAuth2Error>) -> Void) {
+        let types: [Token.RevokeType] = [.accessToken, .refreshToken, .deviceSecret]
+        
+        var errors = [OAuth2Error]()
```

Comment:
Minor comment: Maybe we can remove this mutability by using reduce into.  I was thinking something like this. 
 
```

        let revokeTypes = types.map({ type -> Token.RevokeType? in
            if let revokeType = type.tokenType, token.token(of: revokeType) != nil {
                return type
            }
            return nil
        })
   
        
        let errors = revokeTypes.compactMap({ $0 })
            .reduce(into: [OAuth2Error](), { newErrors, type in
                group.enter()
```

Reply:
Reduce is synchronous, so I'd need to introduce some type of asynchronous `reduce` implementation to ensure the results could be accumulated. Additionally, I'd still want these requests to occur concurrently, which would be difficult to reconcile with `reduce`.

So at this time, without introducing additional dependencies, or requiring Swift Concurrency support, this is probably the cleanest way to accomplish this.

Reply:
Good point, I don't think that would be necessary at all.